### PR TITLE
Add workflow to regenerate code

### PR DESCRIPTION
**Pull request by pshao25**
Add workflow to regenerate code.

User experience:
image
1. Go to github actions from user's fork repo. 
2. Click workflow "Regenerate all code".
3. Choose the branch you are working on, and click "Run workflow".

Result: It helps you run `eng\Generate.ps1` and push a new commit to your branch like:
image

Sample workflow link: https://github.com/pshao25/autorest.csharp/actions/workflows/main.yml
Sample PR link: https://github.com/pshao25/autorest.csharp/pull/1

### CODE REVIEW

File: .github/workflows/regenerate-code.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        shell: pwsh
+        run: |
+          ./eng/Generate.ps1
+
+      - name: Commit generated code
+        run: |
+          git config --global user.name 'actions-user'
+          git config --global user.email 'action@github.com'
+          git add .
+          git commit -m "re-generate test code"
```
Comment (ArcturusZhang) [inline .github/workflows/regenerate-code.yml:38]: maybe we could add the `allow-empty` option here to ensure this command will always succeed and we always see a new commit by github action to distinguish the case that this feature is in a malfunction

**Review by ArcturusZhang — COMMENTED** [review]
This is really helpful, LGTM, only a minor question

File: .github/workflows/regenerate-code.yml (hunk 1 above)
Comment (pshao25) [reply .github/workflows/regenerate-code.yml]: Updated. It will be like
image

**Comment by ArthurMa1978** [comment]
Please run eng/DownloadSharedSource.ps1 to update the shared code.

**Review by ArcturusZhang — APPROVED** [review]

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**